System/Posix/Env/ByteString.hsc

```diff
@@ -116,15 +135,19 @@ foreign import capi unsafe "HsUnix.h unsetenv"
    c_unsetenv :: CString -> IO ()
 # endif
 #else
-unsetEnv name = putEnv (name ++ "=")
+unsetEnv name = putEnv (name <> BC.pack "=")
```

Comment:
This is in fact a compile failure that was never catched

Reply:
```suggestion
unsetEnv name = putEnv (BC.snoc name '=')
```